src/components/modals/modal.tsx

```diff
@@ -0,0 +1,32 @@
+import { PassThrough } from "stream";
+import { ModalProps } from "./index"
+
+
+const Modal = (props: ModalProps) => {
+    const changeState = () => {
+        props.onToggle(false)
+    }
+
+    return (
+        <dialog open={props.isOpen} className=" bg-white border-1-black absolute top-20 left-50">
+            <div className="h-20 pt-4 pb-4">
+                <p className="">{props.title}</p>
+                <form method="dialog">
+                    <ul>
+                        {props.fields.map(field =>
+                            <li key={field}>
+                                <p>{field}</p>
+                                {field.includes("password") && <input type="password" /> || <input type="text" /> }
+                            </li>)}                          
+                    </ul>
+
+                    <button className="p-2" onClick={changeState}>confirm
```

Comment:
Button and changeState function does not belong here. Button should be passed inside children prop.
So as a user of this component I can decide the strategy for closing that modal.